js/common/lib/training-session-impl.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {resolveBackend} from './backend-impl.js';
+import {TrainingSessionHandler} from './backend.js';
+import {InferenceSession as InferenceSession} from './inference-session.js';
+import {processModel, processModelOrOptions} from './session-impl-utils.js';
+import {TrainingSession as TrainingSessionInterface} from './training-session.js';
+
+type SessionOptions = InferenceSession.SessionOptions;
+
+export class TrainingSession implements TrainingSessionInterface {
+  private constructor(handler: TrainingSessionHandler) {
+    this.handler = handler;
+  }
+
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+  /* eslint-disable no-console */
+  runTrainStep(feeds: InferenceSession.OnnxValueMapType, options?: InferenceSession.RunOptions):
+      Promise<InferenceSession.OnnxValueMapType>;
+  runTrainStep(
+      feeds: InferenceSession.OnnxValueMapType, fetches: InferenceSession.FetchesType,
+      options?: InferenceSession.RunOptions): Promise<InferenceSession.OnnxValueMapType>;
+  async runTrainStep(
+      feeds: InferenceSession.OnnxValueMapType, arg1?: InferenceSession.FetchesType|InferenceSession.RunOptions,
+      arg2?: InferenceSession.RunOptions): Promise<InferenceSession.OnnxValueMapType> {
+    // console.log arguments to suppress ts6133 error
+    console.log(feeds);
+    if (arg1) {
+      console.log(arg1);
```

Comment:
I assume this is for debugging when you write the code. finally console.log and `/* eslint-disable no-console */` above should be removed

Reply:
ah, no, I get an error about unused parameters from TypeScript -- this method isn't implemented yet, it's just a placeholder. It might be better to just remove the runTrainStep method completely from this PR though -- I'll do that

Reply:
removed runTrainStep in last commit